homeassistant/components/lutron/__init__.py

```diff
@@ -64,19 +89,25 @@ def setup(hass: HomeAssistant, base_config: ConfigType) -> bool:
         "scene": [],
         "binary_sensor": [],
     }
+    host = config_entry.data[CONF_HOST]
+    uid = config_entry.data[CONF_USERNAME]
+    pwd = config_entry.data[CONF_PASSWORD]
 
-    config = base_config[DOMAIN]
-    hass.data[LUTRON_CONTROLLER] = Lutron(
-        config[CONF_HOST], config[CONF_USERNAME], config[CONF_PASSWORD]
-    )
+    def _connect() -> bool:
+        hass.data[LUTRON_CONTROLLER].load_xml_db()
+        return True
 
-    hass.data[LUTRON_CONTROLLER].load_xml_db()
+    hass.data[LUTRON_CONTROLLER] = Lutron(host, uid, pwd)
```

Comment:
In the config flow we see that we allow more lutron instances. But if we execute this for the second config entry, the first entry is removed. Would it maybe be an idea to store the Lutron object in `[entry.entry_id]`?

Reply:
the current integration only supports a single instance and for now I think we should continue that.  I'm not sure how to restrict that though to avoid the overwrite you are talking about.

Reply:
I've added error checking and the needed translation to return that only a single instance is allowed.